packages/spectral/src/types/Inputs.ts

```diff
@@ -95,7 +96,19 @@ export type ConnectionInput = (
   | TextInputField
   | PasswordInputField
   | BooleanInputField
-) & { shown?: boolean };
+) & {
+  shown?: boolean;
+  /**
+   * Optional value that sets the permission and visibility of the Config Var Input. @default "customer"
+   *
+   * "customer" - Customers can view and edit the Config Var Input.
+   * "embedded" - Customers cannot view or update the Config Var Input as the value will be set programmatically.
+   * "organization" - Customers cannot view or update the Config Var Input as it will always have a default value or be set by the organization.
+   */
+  permissionAndVisibilityType?: PermissionAndVisibilityType;
+  /** Optional value that specifies whether this Config Var Input is visible to an Organization deployer. @default true */
+  visibleToOrgDeployer?: boolean;
+};
```

Comment:
We can't add these types here since these types back component authoring for use in low-code. We support setting these values in low-code on the required config variable. Instead, these need to be added to the IntegrationDefinition types that back CNI. The relevant types are `ConnectionDefinitionConfigVar` and `ConnectionReferenceConfigVar`.

Reply:
![Screen Shot 2024-08-02 at 15 14 13 PM](https://github.com/user-attachments/assets/246daa41-9489-4557-aab0-205858e5e54c)

![Screen Shot 2024-08-02 at 15 13 58 PM](https://github.com/user-attachments/assets/e36ec700-3e52-42ee-8c0e-614a2bff8603)

![Screen Shot 2024-08-02 at 15 17 02 PM](https://github.com/user-attachments/assets/d96b60b9-733c-46a6-80a1-537f21bc1f9f)